File: Benchmarking/src/gen456.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  __attribute__((noinline))
  double f0(const std::vector<s0> &v0, const s0 &v1, const double &v2) {
    const std::vector<std::vector<double>> v6 = v1.p1;
    const std::vector<double> v3 = v6[0];
    const double v11 = v3[0];
    const std::vector<double> v10 = v6[0];
    double v27 = v11;
    const std::vector<double> v24 = v6[0];
    const std::vector<std::vector<double>> v21 { v10, v24, v10, v3, v10 };
    const std::vector<double> v75 = v21[1];
    const double v117 = v75[0];
    const double v148 = v117 * v27;
    return v148;
  }
  int main() {
    std::vector<s0> v0({ { { { 0.0 }, { 1.0 } }, { { 2.0 } } }, { { { 3.0 }, { 4.0 } }, { { 5.0 } } } });
    s0 v1({ { { 6.0 }, { 7.0 } }, { { 8.0 } } });
    double v2(9.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1, v2);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }

File: Benchmarking/src/gen456.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  @inline(never)
  func f0(_ v0: [s0], _ v1: s0, _ v2: Double) -> Double {
    let v6: [[Double]] = v1.p1
    let v3: [Double] = v6[0]
    let v11: Double = v3[0]
    let v10: [Double] = v6[0]
    var v27: Double = v11
    let v24: [Double] = v6[0]
    let v21: [[Double]] = [v10, v24, v10, v3, v10]
    let v75: [Double] = v21[1]
    let v117: Double = v75[0]
    let v148: Double = v117 * v27
    return v148
  }
  func benchmark() {
    let v0: [s0] = [s0(p0: [[0.0], [1.0]], p1: [[2.0]]), s0(p0: [[3.0], [4.0]], p1: [[5.0]])]
    let v1: s0 = s0(p0: [[6.0], [7.0]], p1: [[8.0]])
    let v2: Double = 9.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 9.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
